src/bookshelf/src/bksw_sec.erl

```diff
@@ -35,109 +37,283 @@ is_authorized(Req0, #context{} = Context) ->
+                        end
+                end;
+            _ ->
+                encode_access_denied_error_response(RequestId, Req0, Context)
+        end
+
     catch
-        _:Why ->
-            error_logger:error_report({error, {{mini_s3, make_signed_url_authorization},
-                                               [<<"SECRETKEY">>, Method, Path, Expires, Headers],
-                                               Why}}),
-            error
+        {RequestId, Req0, Context} -> encode_access_denied_error_response(RequestId, Req0, Context)
     end.
 
-do_standard_authorization(RequestId, IncomingAuth, Req0, Context) ->
-    Headers = mochiweb_headers:to_list(wrq:req_headers(Req0)),
-    AmzHeaders = amz_headers(Headers),
-    RawMethod = wrq:method(Req0),
-    Method = string:to_lower(erlang:atom_to_list(RawMethod)),
-    ContentMD5 = proplists:get_value('Content-Md5', Headers, ""),
-    ContentType = proplists:get_value('Content-Type', Headers, ""),
-    Date = proplists:get_value('Date', Headers, ""),
-    %% get_object_and_bucket decodes the bucket, but the request will have been signed with
-    %% the encoded bucket.
-    {ok, Bucket0, Resource} = bksw_util:get_object_and_bucket(Req0),
-    Bucket = bksw_io_names:encode(Bucket0),
-    AccessKey = bksw_conf:access_key_id(Context),
-    SecretKey = bksw_conf:secret_access_key(Context),
-
-    {StringToSign, RawCheckedAuth} =
-        mini_s3:make_authorization(AccessKey, SecretKey,
-                                   erlang:list_to_existing_atom(Method),
-                                   bksw_util:to_string(ContentMD5),
-                                   bksw_util:to_string(ContentType),
-                                   bksw_util:to_string(Date),
-                                   AmzHeaders, bksw_util:to_string(Bucket),
-                                   "/" ++ bksw_util:to_string(Resource),
-                                   ""),
-    CheckedAuth = erlang:iolist_to_binary(RawCheckedAuth),
-    [AccessKeyId, Signature] = split_authorization(IncomingAuth),
-    case erlang:iolist_to_binary(IncomingAuth) of
-        CheckedAuth ->
-            {true, Req0, Context};
+% https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html
+%-spec check_signed_headers_authhead(proplist(), proplist()) -> boolean(). % for erlang20+
+-spec check_signed_headers_authhead(SignedHeaders::[tuple()], Headers::[tuple()]) -> boolean().
+check_signed_headers_authhead(SignedHeaders, Headers) ->
+%    check_signed_headers_common(SignedHeaders, Headers) andalso
+%
+%    % x-amz-content-sha256 header is required
+%    proplists:is_defined("x-amz-content-sha256", SignedHeaders) andalso
+%
+%    % if content-type header is present in request, it is required
+%    case proplists:is_defined("content-type", Headers) of
+%        true ->
+%            proplists:is_defined("content-type", SignedHeaders);
+%        _ ->
+%            true
+%    end.
```

Comment:
We've got a lot of commented code in this file, do we need it?

Reply:
Ahh, these are checks I commented out, related to the temporarily disabled bit you commented on somewhere above.  It's checking that sigv4 headers strictly follow sigv4 specifications.  I need to re-enable things and see what breaks.